packages/nightly/src/lib/nightly.ts

```diff
@@ -0,0 +1,175 @@
+import { isMobile } from "is-mobile";
+import { utils } from "near-api-js";
+import { JsonRpcProvider } from "near-api-js/lib/providers";
+import { createTransaction } from "near-api-js/lib/transaction";
+import { Account, waitFor } from "@near-wallet-selector/core";
+import type { NearNightly, NightlyInjected } from "./injected-nightly";
+import { PublicKey } from "near-api-js/lib/utils";
+
+declare global {
+  interface Window {
+    nightly: NightlyInjected | undefined;
+  }
+}
+
+interface NightlyState {
+  wallet: NearNightly;
+}
+
+const setupNightlyState = (meta: InjectedWalletMetadata): NightlyState => {
+  const wallet = window.nightly?.near;
+  if (!wallet) {
+    window.location.href = meta.downloadUrl;
+    throw new Error("Redirecting to download");
+  }
+  return {
+    wallet: wallet,
+  };
+};
+
+const Nightly: WalletBehaviourFactory<InjectedWallet> = async ({
+  options,
+  store,
+  logger,
+  metadata,
+}) => {
+  const _state = setupNightlyState(metadata);
+  const provider = new JsonRpcProvider({ url: options.network.nodeUrl });
+  const state = store.getState();
+
+  if (state.selectedWalletId === "nightly") {
+    await _state.wallet.connect();
+  }
+  return {
+    // nightly does not support delegating signing right now
+    async signIn() {
+      const existingAccount = _state.wallet.account.accountId;
+
+      if (existingAccount) {
+        const nearAccount: Account = {
+          accountId: _state.wallet.account.accountId,
+        };
+        return [nearAccount];
+      }
+
+      const { accountId } = await _state.wallet.connect();
+      const nearAccount: Account = { accountId };
+      return [nearAccount];
+    },
+
+    async signOut() {
+      await _state.wallet.disconnect();
+    },
+
+    async getAccounts() {
+      const nearAccount: Account = {
+        accountId: _state.wallet.account.accountId,
+      };
+      return [nearAccount];
+    },
+
+    async signAndSendTransaction({ signerId, receiverId, actions }) {
+      logger.log("signAndSendTransaction", { signerId, receiverId, actions });
+      const { contract } = store.getState();
+
+      if (!receiverId && !contract) {
+        throw new Error("Recipient not found");
+      }
+      const blockInfo = await provider.query(
+        `access_key/${_state.wallet.account.accountId}/${
+          _state.wallet.account.publicKey.toString().split(":")[1]
+        }`,
+        ""
+      );
+      const blockHash = utils.serialize.base_decode(blockInfo.block_hash);
+      const tx = createTransaction(
+        signerId || _state.wallet.account.accountId,
+        new PublicKey(_state.wallet.account.publicKey),
+        receiverId || contract!.contractId,
+        // @ts-expect-error
+        ++blockInfo.nonce,
+        actions.map((a) => createAction(a)),
+        blockHash
+      );
+      const signedTransactions = await _state.wallet.signTransaction(tx);
+      const result = await provider.sendTransaction(signedTransactions);
+      return result;
```

Comment:
We should consider using `signTransactions` from our `wallet-utils` package to clean up and standardise this (and `signAndSendTransactions`) logic. See Math Wallet and Ledger implementations - the main change will be implementing a `Signer` interface

Reply:
I don't like implementing Signer. Our API handles only `Transactions`. Implementing `Signer` API will force me to deserialize `Uint8Array` into `Transaction` again also we return already `SignedTransaction` so we will need to cut Signature from it. A lot of changes just to fit into `Signer` schema. It's possible but from my point it will create more code.